lib/crypto/tasks.py

```diff
@@ -24,43 +24,51 @@ def __exit__(self, type, value, traceback):
 
 
 @task
-def sign_addons(addon_ids, **kw):
+def sign_addons(addon_ids, force=False, **kw):
     log.info('[{0}] Signing addons.'.format(addon_ids))
     for version in Version.objects.filter(
             addon_id__in=addon_ids, addon__type=amo.ADDON_EXTENSION):
-        log.info('Signing addon {0}, version {1}'.format(version.addon,
-                                                         version))
-        # We need to bump the version number of the file and the version, so
+        # We need to bump the version number of the file and the Version, so
         # the Firefox extension update mecanism picks this new signed version
         # and installs it.
-        bump_version_number(version)
+        if force:
+            to_sign = version.files.all()
+        else:
+            to_sign = [f for f in version.files.all() if not f.is_signed]
+        if not to_sign:
+            log.info('Not signing addon {0}, version {1} (no files or already '
+                     'signed)'.format(version.addon, version))
+            continue
+        log.info('Signing addon {0}, version {1}'.format(version.addon,
+                                                         version))
         try:
-            version.sign_files()
-        except SigningError as e:
+            for file_obj in to_sign:
+                bump_version_number(file_obj)
+                sign_file(file_obj)
+            # Now update the Version model.
+            version.update(version='{0}.1'.format(version.version))
```

Comment:
So we're updating the version in place? I guess I was thinking that we'd be creating a new version with the extra `{0}.1` version, but that does seem more difficult. I'm wondering how many addon devs would be irked that we took over their version?